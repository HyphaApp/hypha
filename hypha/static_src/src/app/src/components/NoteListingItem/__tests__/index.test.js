import React from "react";
import { mount } from "enzyme";
import sinon from "sinon";
import NoteListingItem from "../index";
import * as enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
enzyme.configure({ adapter: new Adapter() });


describe("Test note listing item component", () => {
    const author = "author text"
    const message = "message text"
    const handleEditNote = jest.fn()
    const timestamp = '20210127'
    const disabled = false
    const editable = true
    const edited = "true"

    const subject = mount(<NoteListingItem
      author={author}
      message={message}
      handleEditNote={handleEditNote}
      timestamp={timestamp}
      disabled={disabled}
      editable={editable}
      edited={edited}
    />);
  
    it("Shoud render without issues", () => {
      expect(subject.length).toBe(1);
    });
  
    it("Should have searched classes and elements", () => {
      expect(subject.find('.note disabled').length).toBe(0);
      expect(subject.find('span').first().text().includes(author)).toBe(true);
      expect(subject.find('.note__edit').length).toBe(1);
      subject.find('a').props().onClick({preventDefault : () =>  1})
      expect(handleEditNote).toHaveBeenCalled();
    });
  
    test('render a note listing item component', () => {
      expect(subject).toMatchSnapshot();
    });
  
  });

  describe("Test note listing item component with author text length greater than 16", () => {
    const author = "author text author text"
    const message = "message text"
    const handleEditNote = jest.fn()
    const timestamp = '20210127'
    const disabled = false
    const editable = true
    const edited = "true"

    const subject = mount(<NoteListingItem
      author={author}
      message={message}
      handleEditNote={handleEditNote}
      timestamp={timestamp}
      disabled={disabled}
      editable={editable}
      edited={edited}
    />);
  
    it("Shoud render without issues", () => {
      expect(subject.length).toBe(1);
    });
  
    it("Should have searched classes and elements", () => {
      expect(subject.find('.disabled').length).toBe(0);
      expect(subject.find('span').first().text().includes("author text auth...")).toEqual(true);
      expect(subject.find('.note__edit').length).toBe(1);
      subject.find('a').props().onClick({preventDefault : () =>  1})
      expect(handleEditNote).toHaveBeenCalled();
    });
  
    test('render a note listing item component with author text length greater than 16', () => {
      expect(subject).toMatchSnapshot();
    });
  
  });

  describe("Test note listing item component with disabled prop", () => {
    const author = "author text"
    const message = "message text"
    const handleEditNote = jest.fn()
    const timestamp = '20210127'
    const disabled = true
    const editable = true
    const edited = "true"

    const subject = mount(<NoteListingItem
      author={author}
      message={message}
      handleEditNote={handleEditNote}
      timestamp={timestamp}
      disabled={disabled}
      editable={editable}
      edited={edited}
    />);
  
    it("Shoud render without issues", () => {
      expect(subject.length).toBe(1);
    });
  
    it("Should have searched classes and elements", () => {
      expect(subject.find('.disabled').length).toBe(1);
      expect(subject.find('span').first().text().includes(author)).toBe(true);
      expect(subject.find('.note__edit').length).toBe(1);
      subject.find('a').props().onClick({preventDefault : () =>  1})
      expect(handleEditNote).toHaveBeenCalled();
    });
  
    test('render a note listing item component with disabled prop', () => {
      expect(subject).toMatchSnapshot();
    });
  
  });
