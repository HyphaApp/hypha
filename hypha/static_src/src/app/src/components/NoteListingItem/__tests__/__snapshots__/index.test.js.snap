// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test note listing item component render a note listing item component 1`] = `
<NoteListingItem
  author="author text"
  disabled={false}
  editable={true}
  edited="true"
  handleEditNote={
    [MockFunction] {
      "calls": Array [
        Array [
          1,
        ],
      ],
    }
  }
  message="message text"
  timestamp="20210127"
>
  <li
    className="note "
  >
    <p
      className="note__meta"
    >
      <span
        className="note__meta note__meta--inner"
      >
        <span>
          author text
        </span>
        <a
          className="note__edit"
          href="#"
          onClick={[Function]}
        >
          Edit
          <svg
            className="icon icon--pen"
          >
            <use
              xlinkHref="#pen"
            />
          </svg>
        </a>
      </span>
      <span
        className="note__date"
      >
        <span
          className="note__date_created"
        >
          20210127
        </span>
        <br />
        <span
          className="note__date_edited"
        >
          (edited: 
          true
          )
        </span>
      </span>
    </p>
    <div
      className="note__content"
      dangerouslySetInnerHTML={
        Object {
          "__html": "message text",
        }
      }
    />
  </li>
</NoteListingItem>
`;

exports[`Test note listing item component with author text length greater than 16 render a note listing item component with author text length greater than 16 1`] = `
<NoteListingItem
  author="author text author text"
  disabled={false}
  editable={true}
  edited="true"
  handleEditNote={
    [MockFunction] {
      "calls": Array [
        Array [
          1,
        ],
      ],
    }
  }
  message="message text"
  timestamp="20210127"
>
  <li
    className="note "
  >
    <p
      className="note__meta"
    >
      <span
        className="note__meta note__meta--inner"
      >
        <span>
          author text auth...
        </span>
        <a
          className="note__edit"
          href="#"
          onClick={[Function]}
        >
          Edit
          <svg
            className="icon icon--pen"
          >
            <use
              xlinkHref="#pen"
            />
          </svg>
        </a>
      </span>
      <span
        className="note__date"
      >
        <span
          className="note__date_created"
        >
          20210127
        </span>
        <br />
        <span
          className="note__date_edited"
        >
          (edited: 
          true
          )
        </span>
      </span>
    </p>
    <div
      className="note__content"
      dangerouslySetInnerHTML={
        Object {
          "__html": "message text",
        }
      }
    />
  </li>
</NoteListingItem>
`;

exports[`Test note listing item component with disabled prop render a note listing item component with disabled prop 1`] = `
<NoteListingItem
  author="author text"
  disabled={true}
  editable={true}
  edited="true"
  handleEditNote={
    [MockFunction] {
      "calls": Array [
        Array [
          1,
        ],
      ],
    }
  }
  message="message text"
  timestamp="20210127"
>
  <li
    className="note disabled"
  >
    <p
      className="note__meta"
    >
      <span
        className="note__meta note__meta--inner"
      >
        <span>
          author text
        </span>
        <a
          className="note__edit"
          href="#"
          onClick={[Function]}
        >
          Edit
          <svg
            className="icon icon--pen"
          >
            <use
              xlinkHref="#pen"
            />
          </svg>
        </a>
      </span>
      <span
        className="note__date"
      >
        <span
          className="note__date_created"
        >
          20210127
        </span>
        <br />
        <span
          className="note__date_edited"
        >
          (edited: 
          true
          )
        </span>
      </span>
    </p>
    <div
      className="note__content"
      dangerouslySetInnerHTML={
        Object {
          "__html": "message text",
        }
      }
    />
  </li>
</NoteListingItem>
`;
